features/ui/select/select.tsx

```diff
@@ -0,0 +1,256 @@
+import { color, textFont, theme } from "@styles/theme";
+import Select, {
+  components,
+  DropdownIndicatorProps,
+  GroupBase,
+  OptionProps,
+  PlaceholderProps,
+  Props,
+  SingleValueProps,
+  StylesConfig,
+} from "react-select";
+import "@fontsource/inter"; // required to pass into react-select component (cannot access otherwise)
+import styled from "styled-components";
+
+// These props MUST be declared at module level to allow custom components and styles to access them via selectProps
+declare module "react-select/dist/declarations/src/Select" {
```

Comment:
Related to the nitpick comment. I'm torn here because this highly couples any code that uses this component to the `react-select` package. If you would decide to replace that package you might need to touch a lot of unrelated code since devs can decide to use any of the `react-select` props directly. So it might make sense to manually define the supported props in this case. This is a bit different from the other components because they use native HTML elements underneath the hood. 

On the other hand it might become annoying to update the manually created types whenever a dev wants to use a feature that's already supported by `react-select`.

Reply:
I completely agree about the disadvantages of highly coupling to `react-select`. Similar to what I discussed in the `disabled` prop comment - is the solution to simply add my own say, `onChange` prop and set that to use the `react-select` `onChange` prop under the hood? That way there is a layer of separation between our own props and the `react-select` props? If so, I think this is a reasonable change to make and not too tedious for new features later.

If that change was made though, the way I see it we *still* have to extend `react-select`'s prop declarations when adding our props. You can't add your own props to the `react-select` actual `Select` component body unless they're already declared at the type definitions module level, otherwise typescript throws an error. For example, adding a `hasError` prop to style the component differently for an error state is impossible otherwise. The `react-select` docs get around this issue using `@ts-ignore` statements in their code snippets, but our codebase doesn't allow that method. I don't see an alternative to this side of things. 

This was the hardest part of completing this task/component - I think a video of how you could handle this one would be extremely useful. Or even taking this discussion to the discord might be good as well?

Reply:
Ah ok, now I get it. It also seems to be the suggested solution in the docs. But obviously that leads to tight coupling (yet more flexibility).

One possible solution is declare the props type separately

```
type SelectProps = {
  label?: string;
  hintMsg?: string;
  error?: boolean;
  errorMsg?: string;
  iconSrc?: string;
  disabled?: boolean;
};

// use SelectProps to define the react-select props
declare module "react-select/dist/declarations/src/Select" {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  export interface Props<
    Option,
    IsMulti extends boolean,
    Group extends GroupBase<Option>
  > extends SelectProps {}
}

export function SelectComponent({
  label,
  hintMsg,
  errorMsg,
  error,
  iconSrc,
  ...Props
// use the own definition of props here
}: SelectProps) {
```

This seems to work. But honestly I'm not sure how clean `export interface Props<...> extends SelectProps {}` is 😄

Reply:
Looks clean enough to me! I'll have a shot converting it to this and see how it goes from here with actually using the Select component elsewhere.